client/packages.go

```diff
@@ -117,3 +118,27 @@ func (client *Client) Snap(name string) (*Snap, error) {
 
 	return pkg, nil
 }
+
+// RemoveSnap removes the snap with the given name
+func (client *Client) RemoveSnap(name string) error {
```

Comment:
Very good point @mvo5! This is the first occurrence in the client library of something that triggers a background operation and returns an async response so we're into new territory here.

Your proposal sounds good to me. Perhaps the thing to do is to return the uuid of the operation here and then eventually introduce `func (client *Client) Operation(uuid string) (*client.Task, error) {}` or similar?